File: world_canvas_client_py/test/test_annotation_collection.py
#!/usr/bin/env python

import roslib; roslib.load_manifest('world_canvas_client_py')
import rospy

import world_canvas_client

if __name__ == '__main__':
    rospy.init_node('objects_loader')
    topic_name  = rospy.get_param('~topic_name', 'annotations')
    topic_type  = rospy.get_param('~topic_type', None)
    pub_as_list = rospy.get_param('~pub_as_list', False)
    world_id = rospy.get_param('~world_id')
    ids      = rospy.get_param('~ids', [])
    names    = rospy.get_param('~names', [])
    types    = rospy.get_param('~types', [])
    keywords = rospy.get_param('~keywords', [])
    related  = rospy.get_param('~relationships', [])

    ac = world_canvas_client.AnnotationCollection(world_id, ids, names, types, keywords, related)
    ac.loadData()

    # Publish annotations' visual markers on client side
    ac.publishMarkers('annotation_markers')

    # Publish annotations on client side
    ac.publish(topic_name + '_client', topic_type, False, pub_as_list)

    # Request server to also publish the same annotations
    ac.publish(topic_name,             topic_type, True,  pub_as_list)

    rospy.loginfo("Done")
    rospy.spin()

File: world_canvas_client_cpp/test/test_annotation_collection.cpp
/*
 * test_annotation_collection.cpp
 *
 *  Created on: Jul 8, 2014
 *      Author: jorge
 */

#include <ros/ros.h>

#include "world_canvas_client_cpp/annotation_collection.hpp"


int main(int argc, char** argv)
{
  ros::init(argc, argv, "test_annotation_collection");

  // Parameters
  ros::NodeHandle nh("~");

  std::string world_id;
  std::string topic_name;
  std::string topic_type;
  std::string default_tn("annotations");
  std::string default_tt;
  std::string default_wi("INVALID_UUID");
  bool        pub_as_list;
  std::vector<std::string> ids;
  std::vector<std::string> names;
  std::vector<std::string> types;
  std::vector<std::string> keywords;
  std::vector<std::string> relationships;

  nh.param("world_id",      world_id, default_wi);
  nh.param("topic_name",    topic_name, default_tn);
  nh.param("topic_type",    topic_type, default_tt);
  nh.param("pub_as_list",   pub_as_list, false);
  nh.param("ids",           ids, ids);
  nh.param("names",         names, names);
  nh.param("types",         types, types);
  nh.param("keywords",      keywords, keywords);
  nh.param("relationships", relationships, relationships);

  // Prepare the annotation collection
  FilterCriteria filter(world_id, ids, names, types, keywords, relationships);
  AnnotationCollection ac(filter);
  ac.loadData();
  ROS_INFO("Annotation collection ready!");

  // Publish annotations' visual markers on client side
  ac.publishMarkers("annotation_markers");

  // Request server to publish the annotations
  ac.publish(topic_name, true, pub_as_list, topic_type);

  ROS_INFO("Done");
  ros::spin();

  return 0;
}
